git_machete/github.py

```diff
@@ -136,6 +137,9 @@ def __fire_github_api_request(method: str, path: str, token: Optional[str], requ
                     first_line + f'You might not have the required permissions for this repository. '
                                  f'Provide a GitHub API token with `repo` access via <b>{GITHUB_TOKEN_ENV_VAR}</b> env var or `gh` or `hub`.\n'
                                  'Visit `https://github.com/settings/tokens` to generate a new one.')
+        elif err.code == http.HTTPStatus.NOT_FOUND:
```

Comment:
Under what circumstances can 404 happen now? 🤔

Reply:
`get_pull_request_by_number_or_none` call for certain pull request number, hence I expect 404 code when given pull requests was never created.

Reply:
Oh! ok I see